daemon/list.go

```diff
@@ -137,6 +138,14 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			Names: names[container.ID],
 		}
 		newC.Image = container.Config.Image
+		if container.Config.Image != "" {
+			img, err := daemon.repositories.LookupImage(container.Config.Image)
```

Comment:
can you explain this change? I'm not sure if we should modify the default behavior here.